posydon/binary_evol/DT/step_detached.py

```diff
@@ -1169,16 +1169,16 @@ def get_star_data(binary, star1, star2, htrack,
             interp1d_pri = get_star_data(
                 binary, primary, secondary, primary.htrack, False)[0]
         else:
-            raise MatchingError("During matching, the primary should either be normal or not normal."
-                                "`non_existent_companion` should be zero.")
+            raise ValueError("During matching, the primary should either be normal (stellar object) or ",
```

Comment:
This is a standard `ValueError`?
I'm not too up-to-date with the warning capturing during a population run.
Does that mean this will stop a POSYDON run?

Reply:
First, it's an error not a warning.
Second, no error in a step can stop POSYDON when running a binary population, because of the try around it with a general exception catch, see [lines 325-341 in binarypopulation.py](https://github.com/POSYDON-code/POSYDON/blob/development/posydon/popsyn/binarypopulation.py)

Reply:
The point/question I was trying to make is that there's a distinctly different treatment of POSYDON errors and general errors in binarypopualtion runs. This should be a POSYDON error, not a standard ValueError, right?
The error before was a POSYDON error.

Reply:
When you just have a test message, POSYDON errors behave the same way as any standard errors. POSYDON errors have the additional ability to deal with POSYDON objects `SingleStar` or `BinaryStar`.
Hence, I'd say here it doesn't make any difference what error type we have. But I suggest to use the one with most appropriate meaning.

Reply:
I changed this from a POSYDON error to a value error because I think it is something that should never happen? So if it does happen the python error should be thrown.

Reply:
Okay! This makes sense. Thank you for the additional clarification :)